### PR TITLE
Oc client limited retries

### PR DESCRIPTION
**Pull request by pbazydlo**
Resolves: https://github.com/opentable/oc/issues/136
Doesn't do exponential backoff.

* Please verify if configuration params make sense.
* When number of retries is exceeded nice error is shown and client stops retrying.
* Retries are counted per component href.

### CODE REVIEW

**Comment by pbazydlo** [comment]
in `test/front-end/ocTestConfig.js` i set `oc.conf.retryInterval = 1` so that retries happen very fast and can be reasonably tested, this is hooked up  in `test/configuration/karma-local.js` so that the config is set before oc-client is loaded. it worked when running `grunt test` and `grunt karma:local` but it seems i missed something

File: test/configuration/karma-common.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+'use strict';
```
Comment (pbazydlo) [inline test/configuration/karma-common.js]: i pulled common configuration from karma-local and karma-sauce into a separate file

File: test/front-end/ocTestConfig.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+var oc = oc || {};
+oc.conf = oc.conf || {};
+oc.conf.retryInterval = 1;
```
Comment (pbazydlo) [inline test/front-end/ocTestConfig.js]: this is needed so i can test number of retries being made before test timeout

**Comment by pbazydlo** [comment]
@matteofigus passing now, as for running karma:sauce locally we could add `karma:sauce-local` task and install chrome driver, then we would be able to run it locally

File: test/front-end/ocTestConfig.js (hunk 2 above)
Comment (matteofigus) [inline test/front-end/ocTestConfig.js]: That's ok, it is exactly how parameters should be set

**Comment by andyroyle** [comment]
looks good to me

**Comment by andyroyle** [comment]
@matteofigus want me to publish?

**Comment by matteofigus** [comment]
@andyroyle that would be cool. When in master, can you also add @pbazydlo here? https://github.com/opentable/oc/blob/master/README.md#contributors

**Comment by pbazydlo** [comment]
yay

**Comment by andyroyle** [comment]
published in 0.20.1

**Comment by matteofigus** [comment]
Thanks :+1:

**Pull request merged**